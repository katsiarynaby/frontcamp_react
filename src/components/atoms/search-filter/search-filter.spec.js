import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import SearchFilter from './index';

describe('SearchFilter', () => {
    it("SearchFilter should render correctly", () => {
        const wrapper = shallow(<SearchFilter
            title="SEARCH BY"
            filters={[{
                key: 'title',
                name: 'title'
            },
            {
                key: 'genres',
                name: 'genres'
            }]}
            activeFilter="title"
            getActiveFilter={jest.fn()}
        />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
