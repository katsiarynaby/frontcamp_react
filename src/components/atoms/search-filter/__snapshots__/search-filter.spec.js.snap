// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter SearchFilter should render correctly 1`] = `
<div
  className="search-filter"
>
  <span
    className="search-filter__title"
  >
    SEARCH BY
  </span>
  <div
    className="search-filter__filters"
  >
    <SearchButton
      active={true}
      key="title"
      onClick={[Function]}
      value="title"
    />
    <SearchButton
      active={false}
      key="genres"
      onClick={[Function]}
      value="genres"
    />
  </div>
</div>
`;
